### PR TITLE
gh-115881: Document feature_version limitations

### PR DESCRIPTION
**Pull request by hauntsaninja**
* Issue: gh-115881

----
📚 Documentation preview 📚: https://cpython-previews--115980.org.readthedocs.build/

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]
This looks great to me!

**Review by pablogsal — APPROVED** [review]

File: Doc/library/ast.rst [hunk 1]
```diff
@@ -2180,12 +2180,13 @@ and classes for traversing abstract syntax trees:
    modified to correspond to :pep:`484` "signature type comments",
    e.g. ``(str, int) -> List[str]``.
 
-   Also, setting ``feature_version`` to a tuple ``(major, minor)``
```
Comment (pablogsal) [inline Doc/library/ast.rst:2183]: I would suggest adding maybe a modal box to highlight this a bit more?

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -2180,12 +2180,13 @@ and classes for traversing abstract syntax trees:
...
-   will attempt to parse using that Python version's grammar.
-   Currently ``major`` must equal to ``3``.  For example, setting
-   ``feature_version=(3, 4)`` will allow the use of ``async`` and
-   ``await`` as variable names.  The lowest supported version is
-   ``(3, 7)``; the highest is ``sys.version_info[0:2]``.
+   Setting ``feature_version`` to a tuple ``(major, minor)`` will result in
+   a "best-effort" attempt to parse using that Python version's grammar.
```
Comment (lysnikolaou) [inline Doc/library/ast.rst:2184]: Maybe also explicitly state that it's not guaranteed to disallow all constructs that weren't possible in a specific version?

Comment (sobolevn) [reply Doc/library/ast.rst:2184]: I would even mention some of them:
- `with` (current issue)
- relaxed grammar for decorators
- maybe something else?

Comment (hauntsaninja) [reply Doc/library/ast.rst:2184]: I pushed a commit that expands what "best-effort" means,  lysnikolaou let me know if that improves the explicitness!

I think I'd prefer not to get into details. I don't want users to expect that we list all divergences in behaviour here, nor do I think listing all divergences would be particularly useful for current users of `feature_version`.

**Review by lysnikolaou — APPROVED** [review]
LGTM!

**Pull request merged**